src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -604,7 +604,9 @@ public static bool ShouldCodeNotBeCompiledIntoFinalImage(InstructionSetSupport i
                         // This instruction set isn't supported on the current platform at all.
                         continue;
                     }
-                    if (instructionSetSupport.IsInstructionSetSupported(instructionSet) || instructionSetSupport.IsInstructionSetExplicitlyUnsupported(instructionSet))
+                    if (instructionSetSupport.IsInstructionSetSupported(instructionSet) ||
+                        instructionSetSupport.IsInstructionSetOptimisticallySupported(instructionSet) ||
```

Comment:
The comment above says "and none of the intrinsic types are in an opportunistic state", so the current behavior looks very intentional and not a bug.

Maybe trigger some optional crossgen2 test runs to see whether the test will identify the problem with this change?

Reply:
A minimal repro to illustrate the problem:
```cs
void Foo()
{
    if (Ssse3.IsSupported)
    {
        Bar();
    }
}

[CompExactlyDependsOn(typeof(Ssse3))]
void Bar()
{
    var a = Ssse3.Shuffle(..);
    ...
}
```

The current behavior for crossgen2 (exactIsaLevel=`Sse2`, opportunisticIsaLevel=`Sse4.2`) is to ignore `Bar()`. But at the same time it's going to compile `Foo` as:
```cs
void Foo()
{
    if (runtimeSsse3Check == true)
    {
        Bar(); // will trigger Tier0 jit compilation
    }
}
```

So feels like we do need to prejit methods demanding opportunistic ISAs. Another option is to lift `[CompExactlyDependsOn(typeof(Ssse3))]` to `Sse2` but that won't work - analyzer will complain on missing `Ssse3` check and also, analyzer is not aware about our opportunistic ISA level so this has to be done in crossgen instead.

cc @davidwrighton who introduced this attribute and the analyzer

Reply:
As far as I remember we can't use Avx as opportunistic with Sse as exact so presumably we can't hit `Vector<>` mismatch problems